test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightOutputContextCongruenceTests.cs

```diff
@@ -0,0 +1,2521 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataJsonLightOutputContextCongruenceTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.OData.Edm;
+using Microsoft.OData.JsonLight;
+using Microsoft.OData.UriParser;
+using Microsoft.Spatial;
+using Xunit;
+
+namespace Microsoft.OData.Tests.JsonLight
+{
+    public class ODataJsonLightOutputContextCongruenceTests
+    {
+        private const string ServiceUri = "http://tempuri.org";
+        private const string batchBoundary = "batch_aed653ab";// Regex used to replace the random batch boundary before equivalence check
+        private const string batchGuidRegex = @"batch[a-z]*_[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}";
+        private EdmModel model;
+        private MemoryStream asyncStream;
+        private MemoryStream syncStream;
+        private ODataMessageWriterSettings writerSettings;
+
+        private EdmEntityType superEntityType;
+        private EdmEntityType orderEntityType;
+        private EdmEntityType customerEntityType;
+        private EdmEntityType orderItemEntityType;
+        private EdmComplexType coordinateComplexType;
+        private EdmEnumType colorEnumType;
+        private EdmEntitySet superEntitySet;
+        private EdmEntitySet orderEntitySet;
+        private EdmEntitySet customerEntitySet;
+        private EdmEntityType streamEntityType;
+        private EdmEntitySet streamEntitySet;
+
+        public ODataJsonLightOutputContextCongruenceTests()
+        {
+            InitializeEdmModel();
+            this.asyncStream = new MemoryStream();
+            this.syncStream = new MemoryStream();
+            this.writerSettings = new ODataMessageWriterSettings
+            {
+                Version = ODataVersion.V4,
+                EnableMessageStreamDisposal = false,
+                BaseUri = new Uri(ServiceUri),
+                ODataUri = new ODataUri { ServiceRoot = new Uri(ServiceUri) }
+            };
+        }
+
+        [Fact]
+        public async Task WriteResourceSet_APIsShouldYieldSameResult()
+        {
+            this.writerSettings.ODataUri.Path = new ODataUriParser(this.model, new Uri(ServiceUri), new Uri(ServiceUri + "/Orders(1)/Items")).ParsePath();
+
+            var orderResourceSet = CreateResourceSet("Orders", "NS.Order");
+            orderResourceSet.Count = 5;

```

Comment:
For request payloads, if we set Count value, we throw an exception. Add a test for that

Reply:
Check out this https://github.com/OData/odata.net/blob/d49b8b9d9a77816b2cf6664be6493ae71906fc0b/src/Microsoft.OData.Core/ODataWriterCore.cs#L1594-L1604

Reply:
@KenitoInc I added a unit tests for that when implementing asynchronous support for `ODataJsonLightWriter` class - https://github.com/OData/odata.net/blob/d49b8b9d9a77816b2cf6664be6493ae71906fc0b/test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightWriterTests.cs#L1640
I have to admit that in this PR I didn't write API comparison tests for all exception cases, at least not as thoroughly as I did in [test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightWriterTests.cs](https://github.com/OData/odata.net/blob/master/test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightWriterTests.cs). It was just too overwhelming if you consider all the exception cases covered in `ODataJsonLightWriterTests` test class. I'll go ahead and add a test for this scenario.